back-end/src/sciler/config/checkConfig.go

```diff
@@ -0,0 +1,367 @@
+// if the config does not follow the manual, a non-empty list of mistakes is returned
+func checkActionLabel(action Action, config WorkingConfig) []string {
+	errorList := make([]string, 0)
+	if _, ok := config.LabelMap[action.TypeID]; ok { // checks if label can be found in the map, if so, it is stored in variable device
+		for _, instruction := range action.Message {
+			for _, comp := range config.LabelMap[action.TypeID] {
+				instruction.ComponentID = comp.ID
+				errorList = append(errorList,
+					checkActionDevice(Action{TypeID: comp.Device.ID, Message: []ComponentInstruction{instruction}}, config)...)
+			}
+		}
+	} else {
+		errorList = append(errorList,
+			fmt.Sprintf("label with id %s not found in map", action.TypeID))
+	}
+	return errorList
+}
+
+// checkValidComparison checks if the comparison is a valid one
+func checkValidComparison(comparison string) bool {
+	comparisonTypesAllowed := []string{"eq", "lt", "gt", "lte", "gte", "contains", "not"}
+	for _, comp := range comparisonTypesAllowed {
+		if comp == comparison {
+			return true
+		}
+	}
+	return false
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (or OrCondition) checkConstraints(config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range or.logics {
+		err := logic.checkConstraints(config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (and AndCondition) checkConstraints(config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range and.logics {
+		err := logic.checkConstraints(config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (condition Condition) checkConstraints(config WorkingConfig, ruleID string) []string {
+	return condition.Constraints.checkConstraints(condition, config, ruleID)
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (or OrConstraint) checkConstraints(condition Condition, config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range or.logics {
+		err := logic.checkConstraints(condition, config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (and AndConstraint) checkConstraints(condition Condition, config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range and.logics {
+		err := logic.checkConstraints(condition, config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
```

Comment:
checkConstraints is a method that checks types and comparator operators =>
checkConstraints is a method that calls the right checker depending on condition.Type